src/domain/users/users.repository.ts

```diff
@@ -71,6 +71,10 @@ export class UsersRepository implements IUsersRepository {
       { user: true },
     );
 
+    if (!wallet.user) {
+      throw new NotFoundException('User not found.');
+    }
+
```

Comment:
Marking as a draft due to these requirements - `Wallet` should require a `User`. This will be updated on the entity. cc @PooyaRaki